src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -2620,6 +2604,14 @@ impl<'a> TypeChecker<'a> {
                     span,
                     lit: Literal::Null,
                 });
+                new_args.push(Expr::Literal {
+                    span,
+                    lit: Literal::Null,
+                });
+                new_args.push(Expr::Literal {
+                    span,
+                    lit: Literal::Null,
+                });
```

Comment:
This is a workaround that ensures there are always at least 3 args for the if function to work properly.

Reply:
before:

```
select coalesce(null, null);
error: APIError: ResponseError with 1065: error:
  --> SQL:1:8
  |
1 | select coalesce(null, null)
  |        ^^^^^^^^^^^^^^^^^^^^ no overload satisfies `if(NULL)`
```

after:

```
select coalesce(null, null);

SELECT
  coalesce(NULL, NULL)

┌──────────────────────┐
│ coalesce(null, null) │
│         Null         │
├──────────────────────┤
│ NULL                 │
└──────────────────────┘
```

Reply:
![image](https://github.com/datafuselabs/databend/assets/36896360/c4e38999-b992-45c1-93d0-cc3d1b91f776)